PublicSDKAPITester/RevenueCatAPI.m

```diff
@@ -0,0 +1,120 @@
+//
+//  RevenueCatAPI.m
+//  MigrateTester
+//
+//  Created by Joshua Liebowitz on 6/18/21.
+//
+
+#import "RevenueCatAPI.h"
+#import "Purchases.h"
+#import "RCPurchases.h"
+
+@import StoreKit;
+
+@implementation RevenueCatAPI
+
+bool canI;
+NSString *version;
+
+BOOL automaticAppleSearchAdsAttributionCollection;
+BOOL debugLogsEnabled;
+RCLogLevel logLevel;
+NSURL *proxyURL;
+BOOL forceUniversalAppStore;
+BOOL simulatesAskToBuyInSandbox;
+RCPurchases *sharedPurchases;
+BOOL isConfigured;
+BOOL allowSharingAppStoreAccount;
+BOOL finishTransactions;
+id<RCPurchasesDelegate> delegate;
+NSString *appUserID;
+BOOL isAnonymous;
+
++ (void)allTheThings {
+//    typedef void (^RCReceivePurchaserInfoBlock)(RCPurchaserInfo * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.ReceivePurchaserInfoBlock);
+//    typedef void (^RCReceiveIntroEligibilityBlock)(NSDictionary<NSString *, RCIntroEligibility *> *) NS_SWIFT_NAME(Purchases.ReceiveIntroEligibilityBlock);
+//    typedef void (^RCReceiveOfferingsBlock)(RCOfferings * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.ReceiveOfferingsBlock);
+//    typedef void (^RCReceiveProductsBlock)(NSArray<SKProduct *> *) NS_SWIFT_NAME(Purchases.ReceiveProductsBlock);
+//    typedef void (^RCPurchaseCompletedBlock)(SKPaymentTransaction * _Nullable, RCPurchaserInfo * _Nullable, NSError * _Nullable, BOOL userCancelled) NS_SWIFT_NAME(Purchases.PurchaseCompletedBlock);
+//    typedef void (^RCDeferredPromotionalPurchaseBlock)(RCPurchaseCompletedBlock);
+//    typedef void (^RCPaymentDiscountBlock)(SKPaymentDiscount * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.PaymentDiscountBlock);
+
+    //    [p presentCodeRedemptionSheet]; ///TODO: iOS ONLY, TEST FOR THIS API BY LOOKING UP SELECTOR
+    RCPurchases *p = [RCPurchases configureWithAPIKey:@""];
+    
+    [RCPurchases setLogHandler:^(RCLogLevel l, NSString *i) {}];
+    canI = [RCPurchases canMakePayments];
+    version = [RCPurchases frameworkVersion];
+    [RCPurchases addAttributionData:@{} fromNetwork:RCAttributionNetworkBranch];
+    [RCPurchases addAttributionData:@{} fromNetwork:RCAttributionNetworkBranch forNetworkUserId:@""];
```

Comment:
I realize it'd be more work, but perhaps we should have these be a method per line? 
that way, if something fails, you know the _exact_ problem just from the method name. 

Then again, we'd already get the compiler error, so we're probably getting the same benefit here.

Reply:
Compiler error is what we're banking on here 😄